inc/zoo/map/RobinHood.h

```diff
@@ -0,0 +1,412 @@
+        auto haystackRichersAsNumber = haystackStrictlyRichers.value();
+        // because we make the assumption of LITTLE ENDIAN byte ordering,
+        // we're interested in the elements up to the first haystack-richer
+        auto deadline = swar::isolateLSB(haystackRichersAsNumber);
+        // to analize before the deadline, "maskify" it.  Remember, the
+        // deadline element itself can't be a potential match, it does
+        // not need preservation
+        auto deadlineMaskified = Metadata{deadline - 1};
+        auto beforeDeadlineMatches = sames & deadlineMaskified;
+        return {
+            deadline,
+            beforeDeadlineMatches
+        };
+    }
+
+    /*! \brief converts the given starting PSL and reduced hash code into a SWAR-ready needle
+    
+    The given needle would have a PSL as the starting (PSL + 1) in the first slot, the "+ 1" is because
+    the count starts at 1, in this way, a haystack PSL of 0 is always "richer"
+    */
+    constexpr static auto makeNeedle(U startingPSL, U hoistedHash) {
+        constexpr auto Ones = meta::BitmaskMaker<U, 1, Width>::value;
+        constexpr auto Progression = Ones * Ones;
+        auto core = startingPSL | (hoistedHash << PSL_Bits);
+        auto broadcasted = broadcast(Metadata(core));
+        auto startingPSLmadePotentialPSLs = Metadata(Progression) + broadcasted;
+        return startingPSLmadePotentialPSLs;
+    }
+
+    template<int Misalignment, typename KeyComparer>
+    constexpr static auto
+    unalignedFind(
+        U hoistedHash,
+        Metadata *base,
+        int homeIndex,
+        const KeyComparer &kc
+    ) noexcept {
+        constexpr auto Ones = meta::BitmaskMaker<U, 1, Width>::value;
+        constexpr auto Progression = Metadata{Ones * Ones};
+
+        MisalignedGenerator<Metadata, 8*Misalignment> p{base};
+        auto index = homeIndex;
+        auto needle = makeNeedle(0, hoistedHash);
+        for(;;) {
+            auto result = potentialMatches(needle, *p);
+            auto positives = result.potentialMatches;
+            auto deadline = result.deadline;
+            while(positives.value()) {
+                auto matchSubIndex = positives.lsbIndex();
+                auto matchIndex = index + matchSubIndex;
+                if(kc(matchIndex)) {
+                    return std::tuple(true, matchIndex);
+                }
+                positives = Metadata{clearLSB(positives.value())};
+            }
+            if(deadline) {
+                auto position = index + Metadata{deadline}.lsbIndex();
+                return std::tuple(false, position);
+            }
+            // Skarupke's tail allows us to not have to worry about the end
+            // of the metadata
+            ++p;
+            index += Metadata::NSlots;
+            needle = needle + Progression;
+        }
+    }
+        
+    template<typename KeyComparer>
+    constexpr auto
+    findThroughIndirectJump(
```

Comment:
why does misalignment only go to 7?

Reply:
0 to 7 assuming a max count of 8 elements (hard-coded for testing purposes)